deploy/osps/default/osp-rhel.yaml

```diff
@@ -571,9 +574,16 @@ spec:
               exit 1
             fi
 
-             # get the full hostname
+            # get the full hostname
             FULL_HOSTNAME=$(hostname -f)
-
+            # if /etc/hostname is not empty then use the hostname from there
+            if [ -s /etc/hostname ]; then
+              FULL_HOSTNAME=$(cat /etc/hostname)
+            fi
+            {{- if eq .CloudProviderName "openstack" }}
+            FULL_HOSTNAME=$(echo $FULL_HOSTNAME | sed -e "s/.novalocal//")
+            {{- end }}
```

Comment:
I don't think this is required here. This should be handled in the bootstrap config instead https://github.com/kubermatic/machine-controller/blob/master/pkg/controller/machine/bootstrap.go#L308

Reply:
I didn't notice that part of code, I'll move this snippet there.

Reply:
It won't work since the second `cloud-init init` restores the original hostname with the `.novalocal` prefix

Reply:
EDIT2: during the test that led me to write the first EDIT, I accidentally didn't update the OSP, therefore there was still the string patch with sed. By removing that one I confirm that it doesn't work since the suffix `.novalocal` is again there after the second `cloud-init init` run.